Modules/Microsoft365DSC/DSCResources/MSFT_AADGroupOwnerConsentSettings/MSFT_AADGroupOwnerConsentSettings.psm1

```diff
@@ -0,0 +1,552 @@
+        }
+
+        if ($PSBoundParameters.ContainsKey('BlockUserConsentForRiskyApps'))
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='BlockUserConsentForRiskyApps'; Value=$BlockUserConsentForRiskyApps})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='BlockUserConsentForRiskyApps';   Value=$currentInstance.BlockUserConsentForRiskyApps})
+        }
+
+        if ($PSBoundParameters.ContainsKey('EnableAdminConsentRequests'))
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='EnableAdminConsentRequests'; Value=$EnableAdminConsentRequests})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='EnableAdminConsentRequests';   Value=$currentInstance.EnableAdminConsentRequests})
+        }
+
+        if ($EnableGroupSpecificConsent -and -not [string]::IsNullOrEmpty($ConstrainGroupSpecificConsentToMembersOfGroupName))
+        {
+            $constrainConsentGroupObj = Get-MgGroup -Filter "DisplayName eq '$ConstrainGroupSpecificConsentToMembersOfGroupName'"
+            if ($null -eq $constrainConsentGroupObj -or $constrainConsentGroupObj.securityEnabled -ne $true)
+            {
+                $message = "ConstrainGroupSpecificConsentToMembersOfGroupName '$ConstrainGroupSpecificConsentToMembersOfGroupName' does not exist or is not a security group"
+                Add-M365DscEvent -Message $message `
+                    -Source $($MyInvocation.MyCommand.Source) `
+                    -EntryType Error `
+                    -EventId 2 `
+                    -EventType Error `
+                    -TenantId $TenantId
+
+                throw $message
+            }
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='ConstrainGroupSpecificConsentToMembersOfGroupId'; Value=$constrainConsentGroupObj.Id})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='ConstrainGroupSpecificConsentToMembersOfGroupId'; Value=$null})
+        }
+
+
+        $BodyParam = @{
+            DisplayName = $templateSettings.DisplayName
+            Values      = $valuesParam
+        }
+
+        if ($currentInstance.Ensure -eq 'Absent')
+        {
+            $BodyParam.Add('TemplateId', $templateSettings.Id)
+            $policy = New-MgGroupSetting @BodyParam
```

Comment:
I have concerns about the structure of the resource. In theory, resources that implement the IsSingleInstance property are tenant wide settings where you cannot create new instance, nor remove any. I believe this resource should be scoped at the group level, which means its primary key should be a group name instead of the Tenant Wide IsSingleInstance.

Reply:
Only one GroupSetting object using this specific template-id can exist at any given time, so in that respect it is Single-Instance.
The GroupSetting object does not necessarily exist *unless* something other than default values have been configured, most likely using the Entra portal. The GroupSettingobject can also be removed again if you want to go back to default values.